### PR TITLE
Fix DeserializeFrom for DisableCompression capability

### LINKED ISSUE #4586: Bug: `NodeCapability.DeserializeFrom` ignores `NodeCapabilityType.DisableCompression` and falls back to `UnknownCapability`
## Description

When deserializing node capabilities, `NodeCapability.DeserializeFrom` does not handle `NodeCapabilityType.DisableCompression`. As a result, any peer advertising the `DisableCompression` capability is treated as having an `UnknownCapability` instead of a proper `DisableCompressionCapability`.

This breaks the intended behavior: nodes that set `Config.EnableCompression = false` (and therefore advertise `DisableCompressionCapability`) will not have that capability correctly recognized by other nodes. Consequently, compression-related decisions based on the capability (e.g. in `VersionPayload`) may be incorrect for incoming peers.

## Expected Behavior

`NodeCapabilityType.DisableCompression` should be deserialized into a `DisableCompressionCapability` instance.

## Actual Behavior

It falls through to the default case and becomes an `UnknownCapability`.

## Proof / Code Reference

The switch in `DeserializeFrom` is missing the `DisableCompression` case:

https://github.com/neo-project/neo/blob/d7eb6ef67eda5e59d19641c5c051524173476a6b/src/Neo/Network/P2P/Capabilities/NodeCapability.cs#L60

```csharp
public static NodeCapability DeserializeFrom(ref MemoryReader reader)
{
    NodeCapabilityType type = (NodeCapabilityType)reader.ReadByte();
    NodeCapability capability = type switch
    {
#pragma warning disable CS0612 // Type or member is obsolete
        NodeCapabilityType.TcpServer or NodeCapabilityType.WsServer => new ServerCapability(type),
#pragma warning restore CS0612 // Type or member is obsolete
        NodeCapabilityType.FullNode => new FullNodeCapability(),
        NodeCapabilityType.ArchivalNode => new ArchivalNodeCapability(),
        _ => new UnknownCapability(type),   // <-- DisableCompression ends up here
    };
    capability.DeserializeWithoutType(ref reader);
    return capability;
}
```

A dedicated class already exists and is used when *sending* the capability:

- `DisableCompressionCapability` – https://github.com/neo-project/neo/blob/d7eb6ef67eda5e59d19641c5c051524173476a6b/src/Neo/Network/P2P/Capabilities/DisableCompressionCapability.cs
- Enum value: `NodeCapabilityType.DisableCompression = 0x03`
- Added by `LocalNode` when compression is disabled:

```csharp
if (!Config.EnableCompression)
{
    capabilities.Add(new DisableCompressionCapability());
}
```

## Suggested Fix

Add the missing case to the switch:

```csharp
NodeCapabilityType.DisableCompression => new DisableCompressionCapability(),
```

## Impact

- Peers that disable compression are not correctly identified.
- Any logic that inspects `capabilities.Any(u => u is DisableCompressionCapability)` (e.g. in `VersionPayload`) will fail for received version payloads.
- The capability is effectively useless on the receiving side even though it is correctly serialized/sent.

### PR DESCRIPTION
**Pull request by cschuchardt88**
# Description

`NodeCapability.DeserializeFrom` did not handle `NodeCapabilityType.DisableCompression`, so peers advertising `DisableCompressionCapability` were deserialized as `UnknownCapability`.

As a result, `VersionPayload.AllowCompression` stayed `true` for those peers and the local node could still send compressed P2P payloads despite the remote requesting no compression (`RemoteNode.SendMessage` → `Message.ToArray(Version.AllowCompression)`).

This PR adds the missing switch case and unit tests covering capability deserialize, `VersionPayload.AllowCompression`, and `Message.ToArray(enablecompression: false)`.

Also includes a small VS Test fix for benchmark projects (`IsTestProject=false` / empty-`TargetFramework` guard) so solution test runs do not hit NETSDK1013.

Closes #4586

# Change Log

- Modified `src/Neo/Network/P2P/Capabilities/NodeCapability.cs` — deserialize `DisableCompression` → `DisableCompressionCapability`
- Added `tests/Neo.UnitTests/Network/P2P/Capabilities/UT_DisableCompressionCapability.cs`
- Extended `tests/Neo.UnitTests/Network/P2P/Payloads/UT_VersionPayload.cs` — `AllowCompression` with/without capability
- Extended `tests/Neo.UnitTests/Network/P2P/UT_Message.cs` — `ToArray` respects compression allow flag
- Modified `benchmarks/Directory.Build.props` — VS Test / empty TargetFramework handling for benchmark projects

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- Unit tests for `DisableCompressionCapability` deserialize round-trip
- Unit tests for `VersionPayload.AllowCompression` create + deserialize
- Unit tests for `Message.ToArray(true/false)` compression strip behavior

- [x] Unit Testing
- [ ] Run Application
- [ ] Local Computer Tests
- [ ] No Testing

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by shargon — DISMISSED** [review]

File: tests/Neo.UnitTests/Network/P2P/UT_Message.cs [hunk 1]
```diff
@@ -183,5 +183,63 @@ public void Compression()
...
+
+            var msg = Message.Create(MessageCommand.Transaction, payload);
+            Assert.IsTrue(msg.IsCompressed, "Test setup requires a compressible Transaction message.");
+
+            var compressed = msg.ToArray(enablecompression: true);
+            var uncompressed = msg.ToArray(enablecompression: false);
+
+            // enablecompression=true keeps the compressed on-wire form.
+            Assert.IsTrue(msg.Flags.HasFlag(MessageFlags.Compressed));
+            Assert.HasCount(30, compressed);
```
Comment (ajara87) [inline tests/Neo.UnitTests/Network/P2P/UT_Message.cs]: Do we need to assert the exact compressed size here? The behavior under test is that enablecompression=false emits the raw payload without the compressed flag. Since the test already verifies that the compressed form is smaller and that the uncompressed form round-trips correctly the exact LZ4 output length makes the test unnecessarily dependent on the compressor implementation

Comment (cschuchardt88) [reply tests/Neo.UnitTests/Network/P2P/UT_Message.cs]: Agreed — the exact compressed byte length is not what this test is trying to lock down.

Dropped `Assert.HasCount(30, compressed)` and kept the checks that matter for the DisableCompression path: compressed form is smaller, `enablecompression=false` clears the compressed flag, and the uncompressed payload still deserializes/round-trips correctly.

**Review by ajara87 — APPROVED** [review]

**Review by shargon — APPROVED** [review]
